syntax = "proto2";
import "common.proto";
import "optype.proto";
package baikaldb.pb;
option cc_generic_services = true;

enum ColumnStatus {
    CS_INVALID = 0;  // 列存没生效
    CS_NORMAL  = 1;  // 列存base文件生效
    CS_COLD    = 2;
};

message ColumnFileInfo {
    optional bytes file_name       = 1;
    optional bytes afs_full_name   = 2;
    optional int64 region_id       = 3;
    optional int64 table_id        = 4;
    optional bool  is_afs_file     = 5;
    optional int64 file_size       = 6;
    optional int64 start_version   = 7;
    optional int64 end_version     = 8;
    optional int32 put_count       = 9;
    optional int32 delete_count    = 10;
    optional int32 merge_count     = 11;
    optional bytes start_key       = 12;
    optional bytes end_key         = 13;
    optional int32 row_count       = 14;
    repeated int64 userids         = 15;
};

message RegionColumnFiles {
    optional int64          max_version    = 1;
    optional ColumnStatus   column_status  = 2;
    repeated ColumnFileInfo active_files   = 3;
    repeated ColumnFileInfo delete_files   = 4;
    optional int64          column_version = 5;
};

message ColumnSparseIndex {
    optional bytes key = 1;
    optional int32 rowgroup_idx = 2;
    optional int64 pos_in_rowgroup = 3;
};

message ColumnMeta {
    repeated ColumnSparseIndex sparse_index = 1;
};
